library/ssl_msg.c

```diff
@@ -3291,10 +3295,59 @@ int mbedtls_ssl_prepare_handshake_record(mbedtls_ssl_context *ssl)
         }
     } else
 #endif /* MBEDTLS_SSL_PROTO_DTLS */
-    /* With TLS we don't handle fragmentation (for now) */
-    if (ssl->in_msglen < ssl->in_hslen) {
-        MBEDTLS_SSL_DEBUG_MSG(1, ("TLS handshake fragmentation not supported"));
-        return MBEDTLS_ERR_SSL_FEATURE_UNAVAILABLE;
+    {
+        int ret;
+        const size_t hs_remain = ssl->in_hslen - ssl->in_hsfraglen;
```

Comment:
Should we check that `ssl->in_hsfraglen <= max_frag_len - 12` before using it? @waleed-elmelegy-arm  what do you think?

Reply:
i don't think it's necessary, here's my understanding: at this point we have message that fit in the buffer and defragmenting can only shrink it, so we won't run out of space. if the client keeps feeding us fragments, we'll run out of buffer space eventually and error out.

Reply:
fair enough, done.